lib/ex_aws/ecs.ex

```diff
@@ -1021,16 +1049,48 @@ defmodule ExAws.ECS do
 
   ## Private parts
 
-  # Pascalize has room for improvements.. but it seems to work
+  defp transform_list(opts, list_key, value_key, fn_transform) do
+    transformed = opts
+    |> Keyword.get(list_key, [])
+    |> transform_list(value_key, fn_transform)
 
-  defp pascalize_keys(opts) when is_map(opts) do
+    Keyword.put(opts, list_key, transformed)
+  end
+  defp transform_list(list, value_key, fn_transform) do
+    list
+    |> Enum.map(&(transform_value(&1, value_key, fn_transform)))
+  end
+
+  defp transform_value(map, key, fn_transform) when is_map(map) do
+    map
+    |> Keyword.new
+    |> transform_value(key, fn_transform)
+    |> Map.new
+  end
+  defp transform_value(opts, key, fn_transform) do
     opts
-    |> Enum.map(&pascalize_keys/1)
-    |> Enum.into(%{})
+    |> Keyword.get_and_update(key, fn(value) ->
```

Comment:
Since the opts transforming are mainly keywords, transform of specific keys are done before normalize.